### DISCUSSION TITLE
Warnings when deploying my provider. DPKeycloakEventListenerProviderFactory is implementing the internal SPI eventsListener.

### DISCUSSION THREAD

**Discussion by PavanKalyan-Gandhapudi** [keycloak/keycloak#41549] [Q&A]
### Description

Currently we are using wildfly keycloak, as a part of this we have SPI Events listener is there. But now we are moving wildfly keycloak to quarkus keycloak. Here when i deploy keycloak-event-listener jar file and build it (using **kc.bat build** command). I was getting this error - 2023-03-09 04:02:37,148 WARN  [org.keycloak.services] (build-37) KC-SERVICES0047: **dp-keycloak-event-listener (com.mf.dp.cell.event.DPKeycloakEventListenerProviderFactory) is implementing the internal SPI eventsListener. This SPI is internal and may change without notice**

After execute kc.bat build command

![image](https://user-images.githubusercontent.com/116148854/226274771-ad44a34e-ad9f-4b2e-9dfb-ac0347b31bb8.png)

**Comment by PavanKalyan-Gandhapudi** [comment]
@dasniko Any comments on this ? 

We are using keycloak-server-spi-private, keycloak-server-spi dependencies.

**Comment by sschu** [comment]
This is just a warning and it means exactly what it says. The interface you are implementing is not public so not guaranteed to be stable. Your SPI might not work with future Keycloak versions and you will have to adapt it.

**Comment by PavanKalyan-Gandhapudi** [comment]
Hi @sschu  Is there any alternate to using internal SPI event listeners ?

**Comment by sschu** [comment]
You can store all events in the database and then poll for events via API. But that might be a lot of event and needs an extra component outside of Keycloak. So implementing a custom listener is probably the natural way to go.

**Comment by dasniko** [comment]
Like @sschu wrote, it's just a warning that you are aware of this. It's not forbidden to use these private interfaces. They are here to be used and implemented.
